staging/src/k8s.io/apiserver/pkg/util/openapi/enablement.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
+
+import (
+	"strings"
+
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+// enumTypeDescriptionHeader is the header of enum section in schema description.
+const enumTypeDescriptionHeader = "Possible enum values:"
+
+// WrapGetOpenAPIDefinitions wraps a GetOpenAPIDefinitions to revert
+// any change to the schema that was made by a disabled feature.
+func WrapGetOpenAPIDefinitions(GetOpenAPIDefinitions common.GetOpenAPIDefinitions) common.GetOpenAPIDefinitions {
+	return func(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
+		defs := GetOpenAPIDefinitions(ref)
+		amendDefinitions(defs)
+		return defs
+	}
+}
+
+func amendDefinitions(defs map[string]common.OpenAPIDefinition) {
+	// revert changes from OpenAPIEnums
+	// TODO(jhf) remove this part with OpenAPIEnums feature gate when it goes GA
```

Comment:
Eventually we'll remove the `genericfeatures.OpenAPIEnums`, I don't know if the TODO is useful.

Reply:
probably not. I will know when to remove this part as soon as the feature gate is dropped.